src/workflows/airqo_etl_utils/bigquery_api.py

```
@@ -630,12 +631,40 @@ def query_data(
+                self.client.query(query=query, job_config=job_config)
+                .result()
+                .to_dataframe()
+            )
+        except google_api_exceptions.GoogleAPIError as e:
+            if isinstance(e, google_api_exceptions.NotFound):
+                logger.error(
+                    f"Query on {table} failed: A specified resource (e.g., dataset or table) was not found. {e.message}"
+                )
+                raise google_api_exceptions.NotFound(
+                    "Query failed: A specified resource (e.g., dataset or table) was not found."
+                )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)